gordo_components/model/models.py

```diff
@@ -786,25 +652,33 @@ def fit(
             )
 
         self._validate_size_of_X(X)
-        gen = self.generate_window(X, output_y=True)
-
-        if not hasattr(self, "model"):
-            # these are only used for the intermediate fit method (fit method of KerasBaseEstimator),
-            # called only to initiate the model
-            super().fit(*next(gen), epochs=1, verbose=0, **kwargs)
-
-        steps_per_epoch = self.calc_steps_per_epoch(X)
 
-        self.kwargs["steps_per_epoch"] = steps_per_epoch
+        # these are only used for the intermediate fit method (fit method of KerasBaseEstimator),
+        # called only to initiate the model
+        # batch_size 1 so we only re-train on a single sample
+        tsg = get_TimeseriesGenerator(
+            data=X, batch_size=1, lookback_window=self.lookback_window, lookahead=1
+        )
+        super().fit(*tsg[0], epochs=1, verbose=0)
```

Comment:
Probably fine. For the cross_val_score it was "ok" already, because the model sent to the cross val was not trained (fit) before that point. And the first step of the cross_val is to clone the model, so that each fold gets a separate thing.